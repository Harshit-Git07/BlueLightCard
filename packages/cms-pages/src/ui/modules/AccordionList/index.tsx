import { cn } from '@/lib/utils';
import Content from '../RichtextModule/Content';

export default function AccordionList({
  intro,
  items,
  layout = 'vertical',
}: Partial<{
  intro?: any;
  items?: Array<{
    summary?: string;
    content?: any;
    open?: boolean;
  }>;
  layout: 'vertical' | 'horizontal';
}>) {
  return (
    <section
      className={cn('section', layout === 'horizontal' ? 'grid gap-8 md:grid-cols-2' : 'space-y-8')}
      itemScope
      itemType="https://schema.org/FAQPage"
    >
      <header
        className={cn(
          'richtext font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark',
          layout === 'horizontal'
            ? 'md:sticky-below-header self-start [--offset:1rem]'
            : 'text-center',
        )}
      >
        <Content value={intro} />
      </header>

      <div className="mx-auto w-full max-w-screen-md grid gap-4">
        {items?.map(({ summary, content, open }, key) => (
          <details
            className="accordion"
            open={open}
            itemScope
            itemProp="mainEntity"
            itemType="https://schema.org/Question"
            key={key}
          >
            <summary
              className="py-[10px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body text-colour-onSurface-light dark:text-colour-onSurface-dark border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"
              itemProp="name"
            >
              {summary}
            </summary>

            <div
              className="anim-fade-to-b mt-2"
              itemScope
              itemProp="acceptedAnswer"
              itemType="https://schema.org/Answer"
            >
              <div
                className="richtext font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark"
                itemProp="text"
              >
                <Content value={content} />
              </div>
            </div>
          </details>
        ))}
      </div>
    </section>
  );
}
