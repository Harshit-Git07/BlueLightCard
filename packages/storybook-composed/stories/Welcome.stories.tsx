import type { Meta, StoryObj } from '@storybook/react';
import WelcomeComponent from './Welcome';

const meta: Meta<typeof WelcomeComponent> = {
  component: WelcomeComponent,
};

export default meta;

type Story = StoryObj<typeof WelcomeComponent>;

export const Welcome: Story = {
  args: {},
  argTypes: {},
};
