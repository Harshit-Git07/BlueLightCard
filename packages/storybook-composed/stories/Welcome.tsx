import React, { FC } from 'react';

const WelcomeComponent: FC = () => (
  <div>
    <p>
      Welcome to the Storybook Composed instance. Here you can find all of our stories across Web,
      Mobile Hybrid, and Shared UI for every brand. Please view these stories from the sidebar.
    </p>
  </div>
);

export default WelcomeComponent;
