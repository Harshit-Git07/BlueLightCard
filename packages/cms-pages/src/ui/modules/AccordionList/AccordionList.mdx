import { Canvas, Controls, Meta, Story, DocsContext } from '@storybook/blocks';
import ComponentStatus from '@bluelightcard/shared-ui/storybook-config/blocks/ComponentStatus';
import FigmaEmbed from '@bluelightcard/shared-ui/storybook-config/blocks/FigmaEmbed';
import * as AccordionList from './AccordionList.stories';

<Meta title="Modules/Accordion List" of={AccordionList} />

# Accordion List

- [Status](#status)
- [Overview](#overview)
- [Props](#props)
- [State](#state)
- [Design](#design)

## Status

<ComponentStatus />

## Overview

Accordion lists are commonly used for things like FAQ's where the title exposes the description in an expanding menu fashion

## Props

<Controls />

## State

This component does not manage any state of its own.

## Design

<FigmaEmbed />

## Variants

List of variants can be shown below here.

### Default

<Canvas of={AccordionList.Default} />

### Horizontal

<Canvas of={AccordionList.Horizontal} />

### Vertical

<Canvas of={AccordionList.Vertical} />
