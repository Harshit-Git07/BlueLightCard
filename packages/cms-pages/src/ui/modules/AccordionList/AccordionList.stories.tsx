import type { AccordionList as AccordionListType } from '@bluelightcard/sanity-types';
import { Meta, StoryObj } from '@storybook/react';
import AccordionList from './';

const intro: AccordionListType['intro'] = [
  {
    children: [
      {
        _type: 'span',
        text: 'Accordion Intro',
        _key: '',
      },
    ],
    style: 'h2',
    _type: 'block',
    _key: '',
  },
];

const items: AccordionListType['items'] = [
  {
    _key: '',
    summary: 'Question 1',
    content: [
      {
        children: [
          {
            _type: 'span',
            text: 'Content for Question 1',
            _key: '',
          },
        ],
        style: 'normal',
        _type: 'block',
        _key: '',
      },
    ],
  },
  {
    _key: '',
    summary: 'Question 2',
    content: [
      {
        children: [
          {
            _type: 'span',
            text: 'Content for Question 2',
            _key: '',
          },
        ],
        style: 'normal',
        _type: 'block',
        _key: '',
      },
    ],
  },
  {
    _key: '',
    summary: 'Question 3',
    content: [
      {
        children: [
          {
            _type: 'span',
            text: 'Content for Question 3',
            _key: '',
          },
        ],
        style: 'normal',
        _type: 'block',
        _key: '',
      },
    ],
  },
];

const meta: Meta<typeof AccordionList> = {
  component: AccordionList,
  title: 'Modules/Accordion List',
  args: {
    intro,
    items,
  },
  parameters: {
    status: 'done',
    design: {
      type: 'figma',
      url: 'https://www.figma.com/embed?embed_host=share&url=https://www.figma.com/design/2bUphiHBRlDyoRoctxwgGC/Static-Landing-Pages?node-id=2143-4328&m=dev',
    },
  },
};

type Story = StoryObj<typeof AccordionList>;

export const Default: Story = {};

export const Horizontal: Story = {
  args: {
    intro,
    items,
    layout: 'horizontal',
  },
};

export const Vertical: Story = {
  args: {
    intro,
    items,
    layout: 'vertical',
  },
};

export default meta;
