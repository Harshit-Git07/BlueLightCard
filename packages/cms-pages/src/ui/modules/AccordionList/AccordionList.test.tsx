import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import * as stories from './AccordionList.stories';

const { Default, Horizontal, Vertical } = composeStories(stories);

describe('Accordion List', () => {
  describe.each([{ Story: Default }, { Story: Horizontal }, { Story: Vertical }])(
    '$Story.storyName',
    ({ Story }) => {
      it('should pass smoke test', () => {
        const result = render(<Story />);
        expect(result).toBeTruthy();
      });

      it('should match snapshot', () => {
        const result = render(<Story />);
        expect(result.container).toMatchSnapshot();
      });
    },
  );
});
