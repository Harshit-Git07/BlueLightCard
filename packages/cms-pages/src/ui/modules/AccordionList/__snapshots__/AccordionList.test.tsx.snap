// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion List Default should match snapshot 1`] = `
<div>
  <section
    class="section space-y-8"
    itemscope=""
    itemtype="https://schema.org/FAQPage"
  >
    <header
      class="richtext font-typography-body-light-weight leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark text-center"
    >
      <div
        class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
      >
        <h2
          class="mb-2 dark:text-colour-onSurface-dark text-typography-headline-bold font-typography-headline-bold-weight leading-typography-headline-bold tracking-typography-headline-bold"
        >
          Accordion Intro
        </h2>
      </div>
    </header>
    <div
      class="mx-auto w-full max-w-screen-md grid gap-4"
    >
      <details
        class="accordion"
        itemprop="mainEntity"
        itemscope=""
        itemtype="https://schema.org/Question"
      >
        <summary
          class="py-[10px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body text-colour-onSurface-light dark:text-colour-onSurface-dark border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"
          itemprop="name"
        >
          Question 1
        </summary>
        <div
          class="anim-fade-to-b mt-2"
          itemprop="acceptedAnswer"
          itemscope=""
          itemtype="https://schema.org/Answer"
        >
          <div
            class="richtext font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark"
            itemprop="text"
          >
            <div
              class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
            >
              <p
                class="mb-2 dark:text-colour-onSurface-dark text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
              >
                Content for Question 1
              </p>
            </div>
          </div>
        </div>
      </details>
      <details
        class="accordion"
        itemprop="mainEntity"
        itemscope=""
        itemtype="https://schema.org/Question"
      >
        <summary
          class="py-[10px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body text-colour-onSurface-light dark:text-colour-onSurface-dark border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"
          itemprop="name"
        >
          Question 2
        </summary>
        <div
          class="anim-fade-to-b mt-2"
          itemprop="acceptedAnswer"
          itemscope=""
          itemtype="https://schema.org/Answer"
        >
          <div
            class="richtext font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark"
            itemprop="text"
          >
            <div
              class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
            >
              <p
                class="mb-2 dark:text-colour-onSurface-dark text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
              >
                Content for Question 2
              </p>
            </div>
          </div>
        </div>
      </details>
      <details
        class="accordion"
        itemprop="mainEntity"
        itemscope=""
        itemtype="https://schema.org/Question"
      >
        <summary
          class="py-[10px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body text-colour-onSurface-light dark:text-colour-onSurface-dark border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"
          itemprop="name"
        >
          Question 3
        </summary>
        <div
          class="anim-fade-to-b mt-2"
          itemprop="acceptedAnswer"
          itemscope=""
          itemtype="https://schema.org/Answer"
        >
          <div
            class="richtext font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark"
            itemprop="text"
          >
            <div
              class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
            >
              <p
                class="mb-2 dark:text-colour-onSurface-dark text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
              >
                Content for Question 3
              </p>
            </div>
          </div>
        </div>
      </details>
    </div>
  </section>
</div>
`;

exports[`Accordion List Horizontal should match snapshot 1`] = `
<div>
  <section
    class="section grid gap-8 md:grid-cols-2"
    itemscope=""
    itemtype="https://schema.org/FAQPage"
  >
    <header
      class="richtext font-typography-body-light-weight leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark md:sticky-below-header self-start [--offset:1rem]"
    >
      <div
        class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
      >
        <h2
          class="mb-2 dark:text-colour-onSurface-dark text-typography-headline-bold font-typography-headline-bold-weight leading-typography-headline-bold tracking-typography-headline-bold"
        >
          Accordion Intro
        </h2>
      </div>
    </header>
    <div
      class="mx-auto w-full max-w-screen-md grid gap-4"
    >
      <details
        class="accordion"
        itemprop="mainEntity"
        itemscope=""
        itemtype="https://schema.org/Question"
      >
        <summary
          class="py-[10px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body text-colour-onSurface-light dark:text-colour-onSurface-dark border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"
          itemprop="name"
        >
          Question 1
        </summary>
        <div
          class="anim-fade-to-b mt-2"
          itemprop="acceptedAnswer"
          itemscope=""
          itemtype="https://schema.org/Answer"
        >
          <div
            class="richtext font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark"
            itemprop="text"
          >
            <div
              class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
            >
              <p
                class="mb-2 dark:text-colour-onSurface-dark text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
              >
                Content for Question 1
              </p>
            </div>
          </div>
        </div>
      </details>
      <details
        class="accordion"
        itemprop="mainEntity"
        itemscope=""
        itemtype="https://schema.org/Question"
      >
        <summary
          class="py-[10px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body text-colour-onSurface-light dark:text-colour-onSurface-dark border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"
          itemprop="name"
        >
          Question 2
        </summary>
        <div
          class="anim-fade-to-b mt-2"
          itemprop="acceptedAnswer"
          itemscope=""
          itemtype="https://schema.org/Answer"
        >
          <div
            class="richtext font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark"
            itemprop="text"
          >
            <div
              class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
            >
              <p
                class="mb-2 dark:text-colour-onSurface-dark text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
              >
                Content for Question 2
              </p>
            </div>
          </div>
        </div>
      </details>
      <details
        class="accordion"
        itemprop="mainEntity"
        itemscope=""
        itemtype="https://schema.org/Question"
      >
        <summary
          class="py-[10px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body text-colour-onSurface-light dark:text-colour-onSurface-dark border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"
          itemprop="name"
        >
          Question 3
        </summary>
        <div
          class="anim-fade-to-b mt-2"
          itemprop="acceptedAnswer"
          itemscope=""
          itemtype="https://schema.org/Answer"
        >
          <div
            class="richtext font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark"
            itemprop="text"
          >
            <div
              class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
            >
              <p
                class="mb-2 dark:text-colour-onSurface-dark text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
              >
                Content for Question 3
              </p>
            </div>
          </div>
        </div>
      </details>
    </div>
  </section>
</div>
`;

exports[`Accordion List Vertical should match snapshot 1`] = `
<div>
  <section
    class="section space-y-8"
    itemscope=""
    itemtype="https://schema.org/FAQPage"
  >
    <header
      class="richtext font-typography-body-light-weight leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark text-center"
    >
      <div
        class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
      >
        <h2
          class="mb-2 dark:text-colour-onSurface-dark text-typography-headline-bold font-typography-headline-bold-weight leading-typography-headline-bold tracking-typography-headline-bold"
        >
          Accordion Intro
        </h2>
      </div>
    </header>
    <div
      class="mx-auto w-full max-w-screen-md grid gap-4"
    >
      <details
        class="accordion"
        itemprop="mainEntity"
        itemscope=""
        itemtype="https://schema.org/Question"
      >
        <summary
          class="py-[10px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body text-colour-onSurface-light dark:text-colour-onSurface-dark border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"
          itemprop="name"
        >
          Question 1
        </summary>
        <div
          class="anim-fade-to-b mt-2"
          itemprop="acceptedAnswer"
          itemscope=""
          itemtype="https://schema.org/Answer"
        >
          <div
            class="richtext font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark"
            itemprop="text"
          >
            <div
              class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
            >
              <p
                class="mb-2 dark:text-colour-onSurface-dark text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
              >
                Content for Question 1
              </p>
            </div>
          </div>
        </div>
      </details>
      <details
        class="accordion"
        itemprop="mainEntity"
        itemscope=""
        itemtype="https://schema.org/Question"
      >
        <summary
          class="py-[10px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body text-colour-onSurface-light dark:text-colour-onSurface-dark border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"
          itemprop="name"
        >
          Question 2
        </summary>
        <div
          class="anim-fade-to-b mt-2"
          itemprop="acceptedAnswer"
          itemscope=""
          itemtype="https://schema.org/Answer"
        >
          <div
            class="richtext font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark"
            itemprop="text"
          >
            <div
              class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
            >
              <p
                class="mb-2 dark:text-colour-onSurface-dark text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
              >
                Content for Question 2
              </p>
            </div>
          </div>
        </div>
      </details>
      <details
        class="accordion"
        itemprop="mainEntity"
        itemscope=""
        itemtype="https://schema.org/Question"
      >
        <summary
          class="py-[10px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body text-colour-onSurface-light dark:text-colour-onSurface-dark border-b border-colour-onSurface-outline-light dark:border-colour-onSurface-outline-dark"
          itemprop="name"
        >
          Question 3
        </summary>
        <div
          class="anim-fade-to-b mt-2"
          itemprop="acceptedAnswer"
          itemscope=""
          itemtype="https://schema.org/Answer"
        >
          <div
            class="richtext font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light text-colour-onSurface-light dark:text-colour-onSurface-dark"
            itemprop="text"
          >
            <div
              class="richtext mx-auto w-full space-y-[1em] [&>:first-child]:!mt-0"
            >
              <p
                class="mb-2 dark:text-colour-onSurface-dark text-typography-body font-typography-body-weight leading-typography-body tracking-typography-body"
              >
                Content for Question 3
              </p>
            </div>
          </div>
        </div>
      </details>
    </div>
  </section>
</div>
`;
